plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateAppEngineStandardMavenProject.java

```diff
@@ -0,0 +1,86 @@
+import org.eclipse.core.runtime.SubMonitor;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.m2e.core.MavenPlugin;
+import org.eclipse.m2e.core.project.ProjectImportConfiguration;
+import org.eclipse.m2e.core.ui.internal.wizards.MappingDiscoveryJob;
+import org.eclipse.ui.actions.WorkspaceModifyOperation;
+
+import java.lang.reflect.InvocationTargetException;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CreateAppEngineStandardMavenProject extends WorkspaceModifyOperation {
+  private static final Logger logger =
+      Logger.getLogger(CreateAppEngineStandardMavenProject.class.getName());
+
+  private AppEngineStandardProjectMavenConfig config;
+
+  public CreateAppEngineStandardMavenProject(AppEngineStandardProjectMavenConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  protected void execute(IProgressMonitor monitor)
+      throws CoreException, InvocationTargetException, InterruptedException {
+    SubMonitor progress = SubMonitor.convert(monitor);
+    monitor.beginTask("Creating Maven AppEngine archetype", 100);
+
+    ProjectImportConfiguration importConfiguration = new ProjectImportConfiguration();
+    IPath creationPath = config.getProject().getParent().getLocation();
+    String groupId = config.getGroupId();
+    String artifactId = config.getArtifactId();
+    String appId = config.getAppEngineProjectId();
+    if (appId == null || appId.trim().isEmpty()) {
+      appId = artifactId;
+    }
+    String appengineArtifactVersion = resolveLatestReleasedArtifact(progress.newChild(20),
+        "com.google.appengine", "appengine-api-1.0-sdk", "jar", "1.9.36");
+    String gcloudArtifactVersion = resolveLatestReleasedArtifact(progress.newChild(20),
+        "com.google.appengine", "gcloud-maven-plugin", "maven-plugin", "2.0.9.106.v20160420");
+
+    Archetype archetype = new Archetype();
+    archetype.setGroupId("com.google.appengine.archetypes");
+    archetype.setArtifactId("appengine-skeleton-archetype");
+    archetype.setVersion("LATEST");
+    Properties properties = new Properties();
+    properties.put("appengine-version", appengineArtifactVersion);
+    properties.put("gcloud-version", gcloudArtifactVersion);
+    properties.put("application-id", appId);
+
+    List<IProject> archetypeProjects = MavenPlugin.getProjectConfigurationManager()
+        .createArchetypeProjects(creationPath, archetype, groupId, artifactId, "0.0.1-SNAPSHOT",
+            config.getPackageName(), properties, importConfiguration, progress.newChild(60));
+
+    Job job = new MappingDiscoveryJob(archetypeProjects);
+    job.schedule();
+  }
+
+  private String resolveLatestReleasedArtifact(SubMonitor progress, String groupId,
+      String artifactId, String type, String defaultVersion) {
+    try {
+      progress.beginTask(MessageFormat.format("Resolving latest version of {0}", artifactId), 10);
```

Comment:
You're going to have to educate me on this one. Why does the string being  user facing imply a need for a MessageFormat?